### DISCUSSION TITLE
Email provider with insecure SMTP for local testing/development

### DISCUSSION THREAD

**Discussion by justinforlenza** [nextauthjs/next-auth#2184] [Help] [answered]
## Question 💬

I am using Next Auth and am trying to setup the email sign in flow. However for my local development, I am using a local SMTP server so I can catch all emails being sent out. (Specifically MailDev)

However I am getting the following error message:
```
https://next-auth.js.org/errors#send_verification_email_error testy@testemail.org Error: self signed certificate
    at TLSSocket.onConnectSecure (_tls_wrap.js:1507:34)
    at TLSSocket.emit (events.js:376:20)
    at TLSSocket._finishInit (_tls_wrap.js:932:8)
    at TLSWrap.ssl.onhandshakedone (_tls_wrap.js:706:12)
    at TLSWrap.callbackTrampoline (internal/async_hooks.js:134:14) {
  code: 'ESOCKET',
  command: 'CONN'
}
```

I do not see any documentation about specifying an unsecure connection. The closest related issue I could find is #807.

I tried setting the `secure: false` in the email provider server config, however, get a typescript error. But even still the `self signed certificate` continues. 

## Steps to reproduce

- Configure Next Email provider to non secured or selfsigned SMTP server
- Try to sign in via email
- Recieve the error

Here is my provider config:
```js
Providers.Email({
      server: {
        host: process.env.EMAIL_SERVER_HOST,
        port: parseInt(process.env.EMAIL_SERVER_PORT),
        auth: {
          user: process.env.EMAIL_SERVER_USER,
          pass: process.env.EMAIL_SERVER_PASSWORD
        },
        secure: false
      },
      from: process.env.EMAIL_FROM
    }),
```

**Comment by justinforlenza** [comment, marked as answer]
So found the solution, had to add the following to the server config: 

``` js
tls: {
 rejectUnauthorized: false
}
```

Found from https://nodemailer.com/smtp/#3-allow-self-signed-certificates

**Reply by justinforlenza** [reply]
However still getting typescript errors, is there any update on #807? 

I know previously the types were managed externally by DefinetlyTyped, but it appears the types have been moved into this repo?

**Reply by balazsorban44** [reply]
They have! If you find any TS issues, please open an issue about it, with a reproduction. Preferably you could also open a PR with the suggested fixes! Thank you! :pray: 

I am guessing the relevant changes have to be made here: https://github.com/nextauthjs/next-auth/blob/main/types/providers.d.ts
